backend/appender/rolling/rolling.go

```diff
@@ -0,0 +1,178 @@
+package rolling
+
+import (
+	"errors"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/urso/ecslog/backend"
+	"github.com/urso/ecslog/backend/layout"
+)
+
+// rolling implements the rolling file appender.
+//
+// TODO: buffered output + buffer flush timeout.
+type rolling struct {
+	lvl backend.Level
+
+	trigger  Trigger
+	strategy Strategy
+	layout   layout.Layout
+
+	mu sync.Mutex
+	wg sync.WaitGroup
+
+	file *os.File
+	stat FileInfo
+}
+
+// Rolloverer is used by the trigger to start the rollover process.
+type Rolloverer interface {
```

Comment:
can we come up with a better name. I think `-er` rule is not always the best option. I would settle with simple `Roller`, `LogRoller`, `FileRoller` or `FileRotator`
This is just a small nit but i think it slightly improves readability

Reply:
I renamed it to 'Rotator'.